File: common-service/src/main/kotlin/ru/romanow/inst/services/common/CommonAutoConfiguration.java
package ru.romanow.inst.services.common;

import org.slf4j.Logger;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import ru.romanow.inst.services.common.properties.ActuatorSecurityProperties;

import javax.annotation.PostConstruct;

import static org.slf4j.LoggerFactory.getLogger;

@Configuration
@ComponentScan("ru.romanow.inst.services.common")
@EnableConfigurationProperties(ActuatorSecurityProperties.class)
public class CommonAutoConfiguration {
    private static final Logger logger = getLogger(CommonAutoConfiguration.class);

    @PostConstruct
    public void postConstruct() {
        logger.info("Common configuration module loaded");
    }
}

File: common-service/src/main/kotlin/ru/romanow/inst/services/common/CommonAutoConfiguration.kt
package ru.romanow.inst.services.common

import jakarta.annotation.PostConstruct
import org.slf4j.LoggerFactory
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.ComponentScan
import ru.romanow.inst.services.common.properties.ActuatorSecurityProperties
import ru.romanow.inst.services.common.properties.CircuitBreakerConfigurationProperties
import ru.romanow.inst.services.common.properties.ServerUrlProperties

@ComponentScan("ru.romanow.inst.services.common")
@EnableConfigurationProperties(
    value = [
        ActuatorSecurityProperties::class,
        CircuitBreakerConfigurationProperties::class,
        ServerUrlProperties::class
    ]
)
class CommonAutoConfiguration {
    private val logger = LoggerFactory.getLogger(CommonAutoConfiguration::class.java)

    @PostConstruct
    fun postConstruct() {
        logger.info("Common configuration module loaded")
    }
}
